docs/HowToSupportTizenRtProtectedBuild.md

```diff
@@ -0,0 +1,343 @@
+- Before starting with protected build, the developer must be aware of the build and boot up process for TizenRT flat build on the intended chipset
+
+### Overview of Protected Build
+- In protected build, two separate binaries namely kernel binary and user binary will be created.
+- After the build process, the kernel binary will be created under the name - tinyara.bin and user binary as - tinyara_user.bin at path <build/output/bin>
+- The kernel binary will consist of the following modules - architecture(ARCH), core Kernel Module(SCHED), C-Lib and memory manager (MM), file system (FS) and the drivers.
+- The user binary will include the following modules  - Applications, C-Lib and Memory Manager (MM), Framework and External modules.
+- C-lib and Memory Manager are duplicated in the kernel space and the user space.
+- As for the memory, the code memory (Flash) region and data memory (RAM) region will be split into two parts each, for use by user and kernel.
+- The kernel and user binaries will be placed in separate code memory regions and during runtime they will use separate data memory regions.
+- The kernel memory regions will be protected by the MPU.
+- When the processor runs in previleged mode, it will have access to all the memory regions. When the processor is running in un-previleged mode, it has access only to user-space memory areas which are configured by the kernel during boot up.
+- User-space code will be able to access the kernel services only through System Calls (downward communication)
+- Information about user-space such as memory regions, and some user-space API's needs to be shared to kernel. This information will be stored in a special data structure called as the User Space Object which will be linked to the kernel binary during build time. The kernel can use the function pointers to user-space APIs available in the user space object to invoke required user space functionality.
+
+
+## Steps for Supporting Protected Build in a new chipset
+### Creating a new configuration for Protected Build
+First,the configuration for protected build should be created. Please follow the steps below:
+- Make a copy of the existing flat build configuration and rename it for protected build.
+	- For example: every chipset has a hello configuration under path build/config/CHIPSET_NAME.
+	- Copy the contents of a config folder say "hello" into a new folder called “hello_protected”
+	- Modify the defconfig and Make.defs as follows:
+
+#### Defconfig
+- The following configs need to be set or unset in the defconfig file.
+
+```
+CONFIG_BUILD_FLAT is not set
+CONFIG_BUILD_PROTECTED=y
+CONFIG_BUILD_2PASS=y
+CONFIG_PASS1_TARGET="all"
+CONFIG_PASS1_OBJECT=""
+CONFIG_TINYARA_USERSPACE=0x08080000
+CONFIG_ARM_HAVE_MPU_UNIFIED=y
+CONFIG_ARMV7M_MPU=y
```

Comment:
MPU is an usage of protected build. I mean that because of protected build, we could protect kernel region using mpu but it is not mandatory of porting protected build. Isn't it?

Reply:
In point of this, creating new configuration is not mandatory. How about separating the mandatory and usages?

Reply:
We also had an assumption that MPU is a usage of protected build. But according to Kconfig, it is mandatory for protected build. 
`depends on ARCH_USE_MPU`

Also, I think that in most cases of protected build, we would like to have the MPU. So, it is better to have these configs as a default setting. Not having an MPU might be an exceptional case.

Reply:
Also, I understand that creating new configuration is not mandatory, but it is the most commonly followed approach when we want to enable protected build for any project. Till now, whenever we have enabled protected build for a new project, we have done it with a new configuration. Adding mandatory and optional cases separately in the document might cause confusion to a new user.

Reply:
ok